aws-datastore/src/test/java/com/amplifyframework/datastore/storage/sqlite/ObserveQueryExecutorTest.java

```diff
@@ -366,27 +367,27 @@ public void observeQueryReturnsSortedListOfTotalItems() throws InterruptedExcept
      * @throws AmplifyException data store exception.
      */
     @Test
+    @Ignore("Failing because of race condition. Added an integration test instead.")
     public void observeQueryReturnsSortedListOfTotalItemsWithInt() throws InterruptedException,
```

Comment:
any dependency on CPK work? If not, can this be done as a separate PR and get it merged in so less changes in this PR?